src/Storages/IStorage.h

```diff
@@ -168,6 +168,25 @@ class IStorage : public std::enable_shared_from_this<IStorage>, public TypePromo
         metadata.set(std::make_unique<StorageInMemoryMetadata>(metadata_));
     }
 
+    /// Daisy: start.
+    /// Get mutable version (snapshot) of storage CreateQuery. CreateQuery object is
+    /// multiversion, so it can be concurrently changed, but returned copy can be
+    /// used without any locks.
+    StorageInMemoryCreateQuery getInMemoryCreateQuery() const { return *create_query.get(); }
```

Comment:
why we need this ?

Reply:
I want to keep the same with the `InMemoryMetadata` interface implementation

Reply:
We don't need to. I think we can have a way more optimized interface by letting `parseCreateQueryFromAST` to return a `StorageInMemoryCreateQueryPtr` and moved it around.